pkg/services/folder/folderimpl/folder.go

```diff
@@ -225,9 +243,83 @@ func (s *Service) GetChildren(ctx context.Context, cmd *folder.GetChildrenQuery)
 		}
 	}
 
+	if cmd.UID == "" {
+		filtered = s.deduplicateAvailableFolders(ctx, filtered)
```

Comment:
instead of looping over all (including direct folders) would it be possible to apply `deduplicateAvailableFolders` only for `availableNonRootFolders`?